### PR TITLE
Add dockerhub release logic and improve main readme

### PR DESCRIPTION
**Pull request by RedLeader962**
# Description

### Summary:

Add the necessary logic to build and push libpointmatcher release docker image to dockerhub. 
Also made some improvment to the main readme:  
- add instruction to use the libpointmatcher release docker image
- add instruction to use the new libpointmatcher installers scripts
- move the summary near the libpointmatcher banner
- move the contributing section to a separate file so that the readme remain easy to consult

Also fixed the new build failure related to pyton binding compilation via github action

### Changes and type of changes (quick overview):

- feat: add dockerhub release logic [NMO-469]
- build: update NBS submodule to latest
- build: add release crawler script and .env file
- docs: move contributing instructions to the `doc/contributing directory` [NMO-442]
  - move contributing instructions to a contributing_instructions.md file
  - move commit_msg_reference.md to `doc/contributing directory` 
  - change remaining ethz-asl domain reference to norlab-ulaval 
  - main `README.MD` 
    - move the summary up, under the top banner 
    - add missing installer script instructions 
    - add docker image usage instructions
- fix: boost download issue >> Move from jfrog artifactory to archives.boost.io
   - see [boost issue #6775](https://github.com/triton-inference-server/server/pull/6775) 
- fix: bump package.xml version to version 1.4.0 [issue #549]
---

# Checklist:

### Code related

- [x] I have made corresponding changes to the documentation 
      (i.e.: function, class, script header, README.md)
- [x] I have commented hard-to-understand code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes 
      (Check [contributing_instructions.md](https://github.com/norlab-ulaval/libnabo/tree/develop/doc/contributing/contributing_instructions.md) for local testing procedure using _libpointmatcher-build-system_)

### PR creation related

- [x] My pull request `base ref` branch is set to the `develop` branch 
     (the _build-system_ won't be triggered otherwise)
- [x] My pull request branch is up-to-date with the `develop` branch 
     (the _build-system_ will reject it otherwise)

### PR description related

- [x] I have included a quick summary of the changes
- [x] I have indicated the related issue's id with `# <issue-id>` if changes are of type `fix`
- [x] I have included a high-level list of changes and their corresponding types
      (See [commit_msg_reference.md](https://github.com/norlab-ulaval/libnabo/tree/develop/doc/contributing/commit_msg_reference.md) for details)

---

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,9 +107,41 @@ and was compiled on:
   * Mac OS X ([see how](/doc/CompilationMac.md))
   * Windows ([see how](/doc/CompilationWindows.md) - partially supported)
 
+### Docker images
+
+Run the following commands to pull and run libnabo in a docker container
```
Comment (boxanm) [inline README.md:112]: Should this be `libnabo` or `libpointmatcher` ?

File: README.md [hunk 2]
```diff
@@ -93,9 +107,41 @@ and was compiled on:
...
+
+```shell
+docker pull norlabulaval/libpointmatcher:latest-ubuntu-focal
+
+docker run -it --rm norlabulaval/libpointmatcher:latest-ubuntu-focal
+```
+
+See
+available [libpointmatcher image tags](https://hub.docker.com/repository/docker/norlabulaval/libpointmatcher/)
```
Comment (boxanm) [inline README.md:121]: The link page currently does not exist on DockerHub

**Review by boxanm — CHANGES_REQUESTED** [review]
Good job Luc! Would be great to add a short guide on how to use the docker image to perform a simple `pmicp` to align two point clouds.

File: README.md (hunk 2 above)
Comment (RedLeader962) [reply README.md:121]: Yes the CI release build needed to finish its job. The image are on dockerhub as of now

**Comment by RedLeader962** [comment]
> Also made some improvment to the main readme:
> 
> - add instruction to use the libpointmatcher release docker image
> - add instruction to use the new libpointmatcher installers scripts
> - move the summary near the libpointmatcher banner
> - move the contributing section to a separate file so that the readme remain easy to consult

@pomerlef I like your approval for the changes that I made to the main [README.md](https://github.com/norlab-ulaval/libpointmatcher/tree/dev-feat-add-dockerhub-release-logic-NMO-469).

Note: The rest of the PR are changes related to release on dockerhub and some build related fix

**Review by simonpierredeschenes — APPROVED** [review]
Everything looks fine, good job!

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -19,7 +20,7 @@
 - [ ] I have commented hard-to-understand code
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] All tests pass locally with my changes 
-      (Check [README.md #Contributing](https://github.com/norlab-ulaval/libpointmatcher/tree/develop#contributing)
+      (Check [contributing_instructions.md](https://github.com/norlab-ulaval/libnabo/tree/develop/doc/contributing/contributing_instructions.md)
```
Comment (boxanm) [inline .github/pull_request_template.md]: The link is pointing to libnabo, shouldn't it be libpointmatcher?

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -33,22 +34,8 @@
...
 - [ ] I have included a quick summary of the changes
 - [ ] I have indicated the related issue's id with `# <issue-id>` if changes are of type `fix`
-- [ ] I have included a high-level list of changes and their corresponding type
-    - Types: `feat` `fix` `docs` `style` `refactor` `perf` `test` `build` `ci` `chore` `revert`
-    - Breaking changes: `<type>!`
-    - Reference:
-      see [commit_msg_reference.md](https://github.com/norlab-ulaval/libpointmatcher/blob/develop/commit_msg_reference.md)
-      in the repository root for details
+- [ ] I have included a high-level list of changes and their corresponding types
+      (See [commit_msg_reference.md](https://github.com/norlab-ulaval/libnabo/tree/develop/doc/contributing/commit_msg_reference.md)
```
Comment (boxanm) [inline .github/pull_request_template.md]: The link is pointing to libnabo. Shouldn't it be libpointmatcher?

File: build_system/README.md [hunk 5]
```diff
@@ -1 +1,2 @@
-See [README.md #Contributing](https://github.com/norlab-ulaval/libpointmatcher/tree/master#contributing) for instructions on how integrate the `libpointmatcher-build-system` to your local development workflow (on your workstation). 
+See [contributing_instructions.md](https://github.com/norlab-ulaval/libnabo/tree/develop/doc/contributing/contributing_instructions.md)
```
Comment (boxanm) [inline build_system/README.md]: The link is pointing to libnabo. Shouldn't it be libpointmatcher?

**Review by boxanm — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md (hunk 3 above)
Comment (RedLeader962) [reply .github/pull_request_template.md]: @boxanm This is outdated. It was fixed in commit [6de0d75](https://github.com/norlab-ulaval/libpointmatcher/pull/550/commits/6de0d7585c6aa5e6283ed53de7482f1790281d52)

File: .github/pull_request_template.md (hunk 4 above)
Comment (RedLeader962) [reply .github/pull_request_template.md]: @boxanm This is outdated. It was fixed in commit [6de0d75](https://github.com/norlab-ulaval/libpointmatcher/pull/550/commits/6de0d7585c6aa5e6283ed53de7482f1790281d52)

File: build_system/README.md (hunk 5 above)
Comment (RedLeader962) [reply build_system/README.md]: @boxanm This is outdated. It was fixed in commit [6de0d75](https://github.com/norlab-ulaval/libpointmatcher/pull/550/commits/6de0d7585c6aa5e6283ed53de7482f1790281d52)

**Review by boxanm — APPROVED** [review]

**Pull request merged**